python/lsst/cp/pipe/cpTask.py

```diff
@@ -0,0 +1,594 @@
+        given a butler for a TS8 (raft-test) repo. It can contain multiple runs, but must correspond to
+        only a single raft/RTM.
+
+        - Run all eotest tasks possible, using the butler to gather the data
+        - Write outputs in eotest format
+
+        In order to replicate the canonical eotest analysis, the tasks should be run in a specific order.
+        This is given/defined in the "Steps" section here:
+        http://lsst-camera.slac.stanford.edu/eTraveler/exp/LSST-CAMERA/
+               displayProcess.jsp?processPath=1179
+
+        But is replicated here for conveniece:
+        * 55Fe Analysis
+        * CCD Read Noise Analysis
+        * Bright Defects Analysis
+        * Dark Defects Analysis
+        * Traps Finding
+        * Dark Current                  X - will not be implemented here
+        * Charge Transfer Efficiencies
+        * Photo-response analysis       X - will not be implemented here
+        * Flat Pairs Analysis
+        * Photon Transfer Curve
+        * Quantum Efficiency            X - will not be implemented here
+
+        List of tasks that exist in the eotest package but aren't mentioned on the above link:
+        --------
+        # linearityTask()
+        # fe55CteTask()
+        # eperTask()
+        # crosstalkTask()
+        # persistenceTask()
+
+        # TODO: For each eotest task, find out what the standard raft testing does for the optional params.
+        i.e. many have optional params for gains, bias-frames etc - if we want bitwise identicallity then we
+        need to know what is typically provided to these tasks when the camera team runs this code.
+        This can probably be worked out from https://github.com/lsst-camera-dh/lcatr-harness
+        but it sounds like Jim Chiang doesn't recommend trying to do that.
+
+        Parameters
+        ----------
+        butler : daf.persistence.butler
+            Butler for the repo containg the eotest data to be used
+        run : string or int
+            Optional run number, to be used for repos containing multiple runs
+        """
+        self.log.info("Running eotest routines direct")
+        runs = butler.queryMetadata('raw', ['run'])
+        if isinstance(run, int):
+            run = str(run)
+        if len(runs) != 1 and run is None:  # lots found and we don't know which one to choose
+            raise RuntimeError("Butler query found %s for runs. eotest datasets must have a run numbers, and"
+                               " must specify which run to use if a respoitory contains multiple runs."%runs)
+        elif run is not None and run not in runs:  # Could be specifying one of many, or one of one here
+            raise RuntimeError("Butler query found %s for runs, but the run specified (%s) "
+                               "was not among them."%(runs, run))
+        elif run is None:  # we know it's OK now
+            run = butler.queryMetadata('raw', ['run'])[0]
```

Comment:
Haven't you already done that query and saved it in `runs`?

Reply:
Yes, but without the `[0]`. Also though, given the comment above, this code no longer exists 🙂